include/wil/resource.h

```diff
@@ -5247,6 +5247,12 @@ namespace wil
             m_globalMemory = GlobalLock(medium.hGlobal);

```

Comment:
Make this constructor delegate to the HGLOBAL constructor to avoid code duplication:
```cpp
unique_hglobal_locked(_In_ STGMEDIUM& medium) : unique_hglobal_locked(medium.hGlobal)
{
}
```

Reply:
Good idea, thanks, done.

Reply:
I have added also constructor taking unique_hglobal& as you suggested.

I do not feel experienced enough to change get() method. It would break existing code if changed to default behaviour.

I have experimented with template<typename T = void>. But I have received only bunch of compiler errors so it is probably over my knowledge, sorry.

Reply:
Don't worry about modifying `get()` or adding templates, I can fix those by myself in another pull request.